### ISSUE TITLE
Regression in enum serialization - only the first instance is handled correctly

### ISSUE THREAD

**Issue by SchmaR** [jsonpickle/jsonpickle#321]
Hi, (I'm using Pyhton 3.8)
Unfortunately, it seems there is a regression in the serialization of enums. When serializing a list of enums, only the first occurrence of an instance is handled correctly. This behavior also holds for enums used in more complex classes. I suspect that there is something wrong when handling the id() identity of objects. The last version that is working as I would expect is 1.2, and the issue persists in any newer version up to 1.4.1.

Here is a minimal setup that fails with 1.4.1
```python
import jsonpickle
from enum import Enum

class Polarity(Enum):
    Present = 1
    Unknown = 0
    Absent = -1

expected = '[{"_value_": -1, "_name_": "Absent", "__objclass__": {"py/type": "__main__.Polarity"}}, {"_value_": -1, "_name_": "Absent", "__objclass__": {"py/type": "__main__.Polarity"}}, {"_value_": 1, "_name_": "Present", "__objclass__": {"py/type": "__main__.Polarity"}}, {"_value_": 1, "_name_": "Present", "__objclass__": {"py/type": "__main__.Polarity"}}]'
print(expected)

actual = jsonpickle.encode([Polarity.Absent, Polarity.Absent, Polarity.Present, Polarity.Present], unpicklable=False)
print(actual)

actual == expected

```
I would expect that the enums are all serialized and that the second instance is not returned as `null` as it happens in the example above.

Output of `print(actual)`:
```json 
[{"_value_": -1, "_name_": "Absent", "__objclass__": {"py/type": "__main__.Polarity"}}, null, {"_value_": 1, "_name_": "Present", "__objclass__": {"py/type": "__main__.Polarity"}}, null]
```

**Comment by davvid** [comment]
The fix from #338 just keeps on giving =)

Please try out the proposed solution there -- this seems to be working correclty with the changes from the `refs` branch in my `davvid` fork.  Please let me know if you see the same.

**Comment by davvid** [comment]
Merged and fixed 😺
